cmd/storer2elastic/elastic/elasticConnector.go

```diff
@@ -0,0 +1,78 @@
+package elastic
+
+import (
+	"fmt"
+
+	"github.com/ElrondNetwork/elrond-go/cmd/storer2elastic/config"
+	"github.com/ElrondNetwork/elrond-go/cmd/storer2elastic/databasereader/disabled"
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/core/indexer"
+	bootstrapDisabled "github.com/ElrondNetwork/elrond-go/epochStart/bootstrap/disabled"
+	"github.com/ElrondNetwork/elrond-go/hashing"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+)
+
+// ConnectorFactoryArgs holds the data needed for creating a new elastic search connector factory
+type ConnectorFactoryArgs struct {
+	ElasticConfig            config.ElasticSearchConfig
+	Marshalizer              marshal.Marshalizer
+	Hasher                   hashing.Hasher
+	ValidatorPubKeyConverter core.PubkeyConverter
+	AddressPubKeyConverter   core.PubkeyConverter
+}
+
+type elasticSearchConnectorFactory struct {
+	elasticConfig            config.ElasticSearchConfig
+	marshalizer              marshal.Marshalizer
+	hasher                   hashing.Hasher
+	validatorPubKeyConverter core.PubkeyConverter
+	addressPubKeyConverter   core.PubkeyConverter
+}
+
+// NewConnectorFactory will return a new instance of a elasticSearchConnectorFactory
+func NewConnectorFactory(args ConnectorFactoryArgs) (*elasticSearchConnectorFactory, error) {
```

Comment:
You created a factory for the indexer :) could have used the method factory pattern here.

Reply:
I created a factory because on the `Create()` function I also init many disabled components and other stuff. Can remain like this in case we want to extend the indexer creation